pdcupdater/utils.py

```diff
@@ -527,7 +527,38 @@ def pkgdb2release(collection):
     return "-".join(release)
 
 
-def tag2release(tag):
+def tag2release(tag, pdc=None):
+    if pdc:
+        return _tag2release_with_pdc(pdc, tag)
+    else:
+        return _tag2release_fedora(tag)
+
+
+def _tag2release_with_pdc(pdc, tag):
+    """ Use PDC to figure out what release is associated with this tag. """
+    releases = list(pdc.get_paged(
+        pdc['releases']._,
+        brew_allowed_tag=tag,
+    ))
+
+    if not releases:
+        raise ValueError("Could not find matching release for tag %r" % tag)
```

Comment:
Hm, this again is intentional.

There's two top-level scenarios:


- We configure pdc-updater to *not* consult PDC about release names, and to instead guess.
- We configure pdc-updater to consult PDC about release names.

  - In the case that pdc-updater is configured to consult PDC, if PDC does not know the name of a tag then there's nothing we can do, and we should abort, error, and alert developers/admins.